From f2112003193b05ec46c76abeab6b2bd7af1d2264 Mon Sep 17 00:00:00 2001
From: Chris Spiegel <cspiegel@gmail.com>
Date: Sun, 10 May 2020 21:01:05 -0700
Subject: [PATCH 1/2] Increase internal linkage in garglk.

Versions of gcc prior to 10 were fine with multiple tentative
definitions, but with 10 they now become linker errors. In fact, these
variables are *not* meant to be related, but instead are meant to be
confined to their own files. It just so happens that some of their names
overlap, so they were being shared. The intent is *not* to share, so
it's correct to give them internal linkage.
---
 garglk/launcher.c  |  8 ++++----
 garglk/launchgtk.c |  6 +++---
 garglk/launchmac.m | 10 +++++-----
 garglk/launchwin.c | 10 +++++-----
 4 files changed, 17 insertions(+), 17 deletions(-)

diff --git a/garglk/launcher.c b/garglk/launcher.c
index de829c3..e2cd064 100644
--- a/garglk/launcher.c
+++ b/garglk/launcher.c
@@ -49,10 +49,10 @@
 #define ID_GLUL (giblorb_make_id('G','L','U','L'))
 
 #define MaxBuffer 1024
-char tmp[MaxBuffer];
-char terp[MaxBuffer];
-char exe[MaxBuffer];
-char flags[MaxBuffer];
+static char tmp[MaxBuffer];
+static char terp[MaxBuffer];
+static char exe[MaxBuffer];
+static char flags[MaxBuffer];
 
 int runblorb(char *path, char *game)
 {
diff --git a/garglk/launchgtk.c b/garglk/launchgtk.c
index bfcf4f1..b69e51b 100644
--- a/garglk/launchgtk.c
+++ b/garglk/launchgtk.c
@@ -41,9 +41,9 @@ static const char * AppName = "Gargoyle " VERSION;
 static const char * LaunchingTemplate = "%s/%s";
 static const char * DirSeparator = "/";
 
-char dir[MaxBuffer];
-char buf[MaxBuffer];
-char tmp[MaxBuffer];
+static char dir[MaxBuffer];
+static char buf[MaxBuffer];
+static char tmp[MaxBuffer];
 
 struct filter
 {
diff --git a/garglk/launchmac.m b/garglk/launchmac.m
index 24ae091..f5951ec 100644
--- a/garglk/launchmac.m
+++ b/garglk/launchmac.m
@@ -40,13 +40,13 @@ static const char * AppName = "Gargoyle " VERSION;
 static const char * LaunchingTemplate = "%s/%s";
 static const char * DirSeparator = "/";
 
-char dir[MaxBuffer];
-char buf[MaxBuffer];
-char tmp[MaxBuffer];
-char etc[MaxBuffer];
+static char dir[MaxBuffer];
+static char buf[MaxBuffer];
+static char tmp[MaxBuffer];
+static char etc[MaxBuffer];
 
 enum FILEFILTERS { FILTER_SAVE, FILTER_TEXT, FILTER_ALL };
-char *winfilters[] =
+static char *winfilters[] =
 {
     "sav",
     "txt",
diff --git a/garglk/launchwin.c b/garglk/launchwin.c
index 893fd68..cdff69f 100644
--- a/garglk/launchwin.c
+++ b/garglk/launchwin.c
@@ -33,11 +33,11 @@ static const char * AppName = "Gargoyle " VERSION;
 static const char * LaunchingTemplate = "\"%s\\%s.exe\" %s \"%s\"";
 static const char * DirSeparator = "\\";
 
-char dir[MaxBuffer];
-char buf[MaxBuffer];
-char tmp[MaxBuffer];
+static char dir[MaxBuffer];
+static char buf[MaxBuffer];
+static char tmp[MaxBuffer];
 
-char filterlist[] =
+static char filterlist[] =
 "All Games\0*.taf;*.agx;*.d$$;*.acd;*.a3c;*.asl;*.cas;*.ulx;*.hex;*.jacl;*.j2;*.gam;*.t3;*.z?;*.l9;*.sna;*.mag;*.dat;*.saga;*.blb;*.glb;*.zlb;*.blorb;*.gblorb;*.zblorb\0"
 "Adrift Games (*.taf)\0*.taf\0"
 "AdvSys Games (*.dat)\0*.dat\0"
@@ -65,7 +65,7 @@ void winmsg(const char *msg)
     MessageBox(NULL, msg, AppName, MB_ICONERROR);
 }
 
-int urldecode(char *decoded, unsigned int maxlen, const char *encoded)
+static int urldecode(char *decoded, unsigned int maxlen, const char *encoded)
 {
     unsigned int i;
     int convert, ascii;
-- 
2.31.1

